pkg/controller/endpoint/endpoints_controller.go

```diff
@@ -402,7 +402,7 @@ func (e *EndpointController) syncService(key string) error {
 				glog.V(5).Infof("Failed to find an IP for pod %s/%s", pod.Namespace, pod.Name)
 				continue
 			}
-			if pod.DeletionTimestamp != nil {
+			if !tolerateUnreadyEndpoints && pod.DeletionTimestamp != nil {
```

Comment:
please augment the comment above the annotation definition with: 
// Endpoints of Services bearing this annotation retain their DNS 
// records and continue receiving traffic for the Service from the moment
// the kubelet starts all containers in the pod and marks it "Running", till the 
// kubelet stops all containers and deletes the pod from the apiserver.